utils/Classes/Panel.js

```diff
@@ -87,10 +87,94 @@ export default class Panel {
         this._elements.delete(id);
     }
 
-    // Switch element
-    /**
-     * @returns {Number}
-     */
+    getElement(id) {
+        return this._elements.get(id);
+    }
+
+    hasElement(id) {
+        return this._elements.has(id);
+    }
+
+    moveZIndexUp(id) {
+        // check if map size allow z-index change
+        if (this._elements.size <= 1) {
+            console.log("Can't change the z-index when only one element is in the panel.");
+            return;
+        }
+
+        let element = this.getElement(id);
+        let elementZIndex = element.currentState().z;
+
+        // check if z-index the highest
+        if (elementZIndex === this.getHighestZIndex()) {
+            console.log('the element is already the highest z-index of the panel.');
+            return;
+        }
+
+        // return closest higher Z-Index element
+        let zIndex = this.getHighestZIndex();
+        let topElement = null;
+        // look for z > elementZ starting from highest
+        this._elements.forEach((el) => {
+            if (el.currentState().z <= zIndex && el.currentState().z > elementZIndex) {
+                zIndex = el.currentState().z;
```

Comment:
We did say, that from now on we use Getter and Setter.

Reply:
function dated from before the changes. I will change.